core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -3633,6 +3634,18 @@ protected void addInitialAdminData(boolean addAdmin, boolean initialSetup) throw
                     log.error(message);
                 }
             } else if (addAdmin) {
+                if (!checkUserNameValid(adminUserName)) {
```

Comment:
Rather duplicating the code, can you extract that logic to a private method as mentioned in [1]?

[1] https://github.com/wso2/carbon-kernel/pull/517/files#diff-ce9fb4f89114a5caaeda8d45be79d46eR3714